pika/adapters/base_connection.py

```diff
@@ -123,6 +123,8 @@ def _adapter_connect(self):
         for sock_addr in addresses:
             error = self._create_and_connect_to_socket(sock_addr)
             if not error:
+                # Make the socket non-blocking after the connect
+                self.socket.setblocking(0)
```

Comment:
The whole connect sequence including SSL handshake is still done with a blocking socket. I have a separate sequence of patches that makes the whole lot non-blocking (which I think is cleaner) but it requires quite a lot of fixing to Select and Asyncore Connections.